### ISSUE TITLE
Support for AbortSignal

### ISSUE THREAD

**Issue by timdp** [grantila/fetch-h2#35]
So I'd like to make my `fetch` calls cancelable by passing an `AbortSignal`.

The fetch-h2 docs link to [`AbortSignal`](https://developer.mozilla.org/docs/Web/API/AbortSignal), suggesting that the `signal` option is supported. However, looking at the code:

- I only see a few statements that synchronously check `signal.aborted`. That's only part of the API. I was also expecting a `signal.addEventListener('abort', fn)` so that `fn` gets triggered when `AbortController#abort()` is called.

- Neither `AbortSignal` nor `AbortController` is actually exposed by the package. I suppose I could polyfill them.

Long story short, I guess this feature is not supported yet? :slightly_smiling_face:

**Comment by grantila** [comment]
Right, this is only used internally now to implement timeout. I'm building full support as I write this.

**Comment by timdp** [comment]
Awesome! :+1:
